my-app/src/app/main-layout/content-projection/course/course.component.ashost.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, OnInit } from '@angular/core';
+import { ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { Course } from 'src/app/models/course';
+
+import { CourseComponent } from './course.component';
+
+describe('CourseComponent', () => {
+
+  let hostComponent: TestHostComponent;
+  let hostFixture: ComponentFixture<TestHostComponent>;
+  const courseContent = new Course(1, 'hosted-course', new Date(), 100, 'hosted-course-description');
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [ CourseComponent, TestHostComponent ]
+    })
+    .compileComponents();
+  });
+
+  beforeEach(() => {
+
+    hostFixture = TestBed.createComponent(TestHostComponent);
+    hostComponent = hostFixture.componentInstance;
+    hostFixture.detectChanges();
+
+  });
+
+  it('should receive course as input from host', () => {
+    expect(hostFixture.nativeElement.querySelector('app-course').innerText).toContain(courseContent.title);
+  });
+
+  it('should remvove app-course component from host when delete is clicked', fakeAsync(() => {
+    spyOn(hostComponent, 'onDeleteCourse');
+    const s = hostFixture.debugElement.nativeElement.querySelector('#delete-button');
```

Comment:
try to name variables with more obvious names and give type for each

Reply:
sure